lib/db_utils.go

```diff
@@ -4670,109 +4670,129 @@ func DeleteUtxoOperationsForBlockWithTxn(txn *badger.Txn, snap *Snapshot, blockH
 	}
 
 	// Height
 	height, err := ReadUvarint(rr)
 	if err != nil {
-		return nil, errors.Wrapf(err, "DeserializeBlockNode: Problem decoding Height")
+		return errors.Wrapf(err, "DeserializeBlockNode: Problem decoding Height")
 	}
 	blockNode.Height = uint32(height)
 
-	// DifficultyTarget
-	_, err = io.ReadFull(rr, blockNode.DifficultyTarget[:])
-	if err != nil {
-		return nil, errors.Wrapf(err, "DeserializeBlockNode: Problem decoding DifficultyTarget")
-	}
+	if !MigrationTriggered(blockHeight, ProofOfStake2ConsensusCutoverMigration) {
+		// DifficultyTarget
+		_, err = io.ReadFull(rr, blockNode.DifficultyTarget[:])
+		if err != nil {
+			return errors.Wrapf(err, "DeserializeBlockNode: Problem decoding DifficultyTarget")
+		}
 
-	// CumWork
-	tmp := BlockHash{}
-	_, err = io.ReadFull(rr, tmp[:])
-	if err != nil {
-		return nil, errors.Wrapf(err, "DeserializeBlockNode: Problem decoding CumWork")
+		// CumWork
+		tmp := BlockHash{}
+		_, err = io.ReadFull(rr, tmp[:])
+		if err != nil {
+			return errors.Wrapf(err, "DeserializeBlockNode: Problem decoding CumWork")
+		}
+		blockNode.CumWork = HashToBigint(&tmp)
 	}
-	blockNode.CumWork = HashToBigint(&tmp)
 
 	// Header
 	payloadLen, err := ReadVarint(rr)
 	if err != nil {
-		return nil, errors.Wrapf(err, "DeserializeBlockNode: Problem decoding Header length")
+		return errors.Wrapf(err, "DeserializeBlockNode: Problem decoding Header length")
 	}
 	headerBytes, err := SafeMakeSliceWithLength[byte](uint64(payloadLen))
 	if err != nil {
-		return nil, errors.Wrapf(err, "DeserializeBlockNode: Problem cretaing byte slice for header bytes")
+		return errors.Wrapf(err, "DeserializeBlockNode: Problem cretaing byte slice for header bytes")
 	}
 	_, err = io.ReadFull(rr, headerBytes[:])
 	if err != nil {
-		return nil, errors.Wrapf(err, "DeserializeBlockNode: Problem reading Header bytes")
+		return errors.Wrapf(err, "DeserializeBlockNode: Problem reading Header bytes")
 	}
 	blockNode.Header = NewMessage(MsgTypeHeader).(*MsgDeSoHeader)
 	err = blockNode.Header.FromBytes(headerBytes)
 	if err != nil {
-		return nil, errors.Wrapf(err, "DeserializeBlockNode: Problem parsing Header bytes")
+		return errors.Wrapf(err, "DeserializeBlockNode: Problem parsing Header bytes")
 	}
 
 	// Status
 	status, err := ReadUvarint(rr)
 	if err != nil {
-		return nil, errors.Wrapf(err, "DeserializeBlockNode: Problem decoding Status")
+		return errors.Wrapf(err, "DeserializeBlockNode: Problem decoding Status")
 	}
 	blockNode.Status = BlockStatus(uint32(status))
 
+	// CommittedStatus
+	if MigrationTriggered(blockHeight, ProofOfStake2ConsensusCutoverMigration) {
+		committedStatus, err := ReadUvarint(rr)
+		if err != nil {
+			return errors.Wrapf(err, "DeserializeBlockNode: Problem decoding CommittedStatus")
+		}
+		blockNode.CommittedStatus = CommittedBlockStatus(committedStatus)
+	}
+	return nil
+}
+
+func SerializeBlockNode(blockNode *BlockNode) ([]byte, error) {
```

Comment:
Where will this be used? Would it make sense to call serializeBlockNode(blockNode, blockHeight) there instead?